src/core/exec-credential.c

```diff
@@ -140,6 +178,37 @@ int exec_context_put_set_credential(
         return 0;
 }
 
+int exec_context_put_import_credential(ExecContext *c, const char *glob, const char *rename) {
+        _cleanup_(exec_import_credential_freep) ExecImportCredential *ic = NULL;
+        int r;
+
+        assert(c);
+
+        ic = new(ExecImportCredential, 1);
+        if (!ic)
+                return -ENOMEM;
+
+        *ic = (ExecImportCredential) {
+                .glob = strdup(glob),
+                .rename = !isempty(rename) ? strdup(rename) : NULL,
+        };
+        if (!ic->glob || (rename && !ic->rename))
```

Comment:
rename here needs isempty too? Maybe, normalize with `rename = empty_to_null(rename)` in the beginning of the func?